### PR TITLE
Added intersect-rect-and-line support for SDL_IntersectRectAndLine()

### PR DESCRIPTION
**Pull request by ellisvelo**
Hello,

I've added a function into rect.lisp that adds intersection checking with a rectangle and a line. Thanks for your support of cl-sdl2.

### CODE REVIEW

File: src/rect.lisp [hunk 1]
```diff
@@ -236,6 +236,21 @@ value is always a newly allocated SDL_Rect structure."
         (return-from intersect-rect (values nil empty))))
     (values t intersect)))
 
+(defun intersect-rect-and-line (rect x1 y1 x2 y2)
+  "Returns five values where the first value is T if the coordinates of the line
+intersect RECT. The remaining returned values represent the starting and ending
+coordinates of the line clipped to the boundary of the rectangle."
+  (c-with ((x1pos :int) (y1pos :int)
+	   (x2pos :int) (y2pos :int))
```
Comment (mfiano) [inline src/rect.lisp:244]: This should be on the previous line.

File: src/rect.lisp [hunk 2]
```diff
@@ -236,6 +236,21 @@ value is always a newly allocated SDL_Rect structure."
...
+
+    (setf x1pos x1 y1pos y1
```
Comment (mfiano) [inline src/rect.lisp:246]: I would argue this is harder to read than each lhs/rhs on its own line.

Comment (mfiano) [reply src/rect.lisp:246]: This should be indented properly and in a format as per the above comment.

Comment (mfiano) [inline src/rect.lisp:246]: I would argue this is harder to read than each lhs/rhs on its own line.

File: src/rect.lisp [hunk 3]
```diff
@@ -236,6 +236,21 @@ value is always a newly allocated SDL_Rect structure."
...
+	  x2pos x2 y2pos y2)
```
Comment (mfiano) [inline src/rect.lisp:247]: This should be indented properly and in a format as per the above comment.

File: src/rect.lisp [hunk 4]
```diff
@@ -236,6 +236,21 @@ value is always a newly allocated SDL_Rect structure."
...
+
+    (let ((intersected
+	    (sdl-true-p (sdl-intersect-rect-and-line rect (x1pos &) (y1pos &)
+						     (x2pos &) (y2pos &)))))
```
Comment (mfiano) [inline src/rect.lisp:251]: Bad indentation here as well

**Review by mfiano — CHANGES_REQUESTED** [review]
Thanks. Please fix the requested code changes and I'll get to merging this after testing.

File: src/rect.lisp (hunk 2 above)
Comment (ellisvelo) [reply src/rect.lisp]: Not a problem. I was originally trying to follow the same style as mouse-state and get-global-mouse-state.

File: src/rect.lisp (hunk 3 above)
Comment (ellisvelo) [reply src/rect.lisp]: Done

File: src/rect.lisp (hunk 4 above)
Comment (ellisvelo) [reply src/rect.lisp]: I aligned the code on the same line, but the line seems a little long.

**Comment by Halfwake** [comment]
This should be good to merge now, right?

**Pull request merged**